src/common/captcha_.py

```
@@ -1,14 +1,23 @@
 
+from streamlit.source_util import page_icon_and_name
 from captcha.image import ImageCaptcha
+from pathlib import Path
 
+import hashlib
 import random
 import string
 import os
 
+def calc_md5(string : str):
+    return hashlib.md5(string.encode()).hexdigest() 
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)